lib/routes/routes.dart

```diff
@@ -365,63 +366,74 @@ class AppRouter extends RouterDelegate<AppRoutes<Object?>>
 
   @override
   Widget build(BuildContext context) {
-    return _AppRouterBuilder(
-      builder: (context) {
-        final pages = <Page<void>>[
-          StackFadePage(
-            key: AppRoutes.initial.uniqueKey,
-            child: const InitialRoute(),
-            transitionSettings: transitionSettings.initial,
-          ),
-        ];
-
-        for (int i = 0; i < _routes.length; i++) {
-          final route = _routes[i];
-          if (route.hasSameLocation(AppRoutes.settings)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.settings.uniqueKey,
-              child: const SettingsRoute(),
-              transitionSettings: transitionSettings.dismissible,
-            ));
-          } else if (route.hasSameLocation(AppRoutes.generalSettings)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.generalSettings.uniqueKey,
-              child: const GeneralSettingsRoute(),
-              transitionSettings: transitionSettings.theme,
-            ));
-          } else if (route.hasSameLocation(AppRoutes.themeSettings)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.themeSettings.uniqueKey,
-              child: const ThemeSettingsRoute(),
-              transitionSettings: transitionSettings.theme,
-            ));
-          } else if (route.hasSameLocation(AppRoutes.licenses)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.licenses.uniqueKey,
-              child: const LicensePage(),
-              transitionSettings: transitionSettings.dismissible,
-            ));
-          } else if (route.hasSameLocation(AppRoutes.dev)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.dev.uniqueKey,
-              child: const DevRoute(),
-              transitionSettings: transitionSettings.dismissible,
-            ));
-          } else if (route.hasSameLocation(AppRoutes.selection)) {
-            pages.add(StackFadePage(
-              key: AppRoutes.selection.uniqueKey,
-              child: SelectionRoute(
-                selectionArguments: (route as AppRoutes<SelectionArguments>).arguments!,
-              ),
-              transitionSettings: transitionSettings.greyDismissible,
-            ));
-          }
+    return Consumer(
+      builder: (context, ref, child) {
```

Comment:
Because the `MediaQuery`, which this `Consumer` needs to use, is no longer available outside of the `MaterialApp.router`.
See the commit message of 57efb33d1280e9fef3947a69ab9f0209ee3148dd for context.

Reply:
It was previously working pretty fine with use `useInheritedMediaQuery: true`.

I see that the method I used to create the new `MediaQuery` is not entirely correct, but we could use `MediaQuery.fromWindow` instead.

So I would prefer that you return the `MediaQueryWrapper`, and do this `MediaQuery.fromWindow -> Builder -> MediaQuery(data: MediaQuery.of(context).copyWith(textScaleFactor: textScaleFactor), child: child)`.

I don't agree with mixing this into the router class.

Reply:
Ok, I didn't see `MediaQuery.fromWindow`, that is a good option.